compiler/modules/front/expr.bal

```diff
@@ -870,12 +870,57 @@ function codeGenListConstructor(ExprContext cx, bir:BasicBlock bb, t:SemType? ex
         { result: operand, block: nextBlock } = check codeGenExprForType(cx, nextBlock, requiredType, member, "incorrect type for list member");
         operands.push(operand);
     }
+    check fillListConstructOperands(cx, nextBlock, operands, atomicType, resultType, expr);
     bir:TmpRegister result = cx.createTmpRegister(resultType, expr.opPos);
     bir:ListConstructInsn insn = { operands: operands.cloneReadOnly(), result, pos: expr.opPos };
     nextBlock.insns.push(insn);
     return { result, block: nextBlock };
 }
 
+function fillListConstructOperands(ExprContext cx, bir:BasicBlock bb, bir:Operand[] operands, 
+                                   t:ListAtomicType atomicType, t:SemType semType, s:ListConstructorExpr expr) returns ResolveTypeError? {
+    int fixedLength = atomicType.members.fixedLength;
+    if operands.length() >= fixedLength {
+        return;
+    }
+    t:Filler? filler = t:filler(cx.mod.tc, semType); 
+    // we currently don't support optional fields for mapping types
+    if filler !is t:ListFiller {
+        return cx.semanticErr("no filler value", s:range(expr));
+    }
+    t:Filler[] memberFillers = filler.memberFillers;
+    t:Filler rest = memberFillers[memberFillers.length() - 1]; 
+    foreach int i in operands.length() ..< fixedLength {
+        t:Filler operandFiller = i < memberFillers.length() ? memberFillers[i] : rest;
+        operands.push(fillerToOperand(cx, bb, operandFiller, expr));
+    }
+}
+
+function fillerToOperand(ExprContext cx, bir:BasicBlock bb, t:Filler filler, s:ListConstructorExpr expr) returns bir:Operand {
```

Comment:
Functions that add insns to a basic block are called codeGenXXX
xxxToOperand is not a good name for a function with side effects.

Reply:
fixed